src/parser.cpp

```diff
@@ -565,8 +565,21 @@ job_t *parser_t::job_get(job_id_t id) {
 job_t *parser_t::job_get_from_pid(pid_t pid) {
     job_iterator_t jobs;
     job_t *job;
+
+    pid_t pgid = getpgid(pid);
+
+    if (pgid < 0) {
```

Comment:
FWIW, I prefer `if (pgid == -1) {` since the syscall in question will never return a different negative value on error. I mention this only because there our code is inconsistent about this and I've been slowly switching things to the `retval == -1` pattern.